### PR TITLE
Add 2d grid sampler

### PR DESCRIPTION
**Pull request by DenisVieriu97**

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/GridSampler.mm [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        CachedGraph *newCachedGraph = nil;
+        @autoreleasepool {
+          MPSGraph* mpsGraph = make_mps_graph();
+          newCachedGraph = new CachedGraph(mpsGraph);
+
+          MPSGraphTensor* inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, input);
+          MPSGraphTensor* gridTensor = mpsGraphRankedPlaceHolder(mpsGraph, grid);
+
+          MPSGraphTensor* outputTensor = [mpsGraph sampleGridWithSourceTensor: inputTensor
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/GridSampler.mm:74]: @DenisVieriu97 , you will need to add check for `respondsToSelector` here. We can use Ventura 13.1 or something, but i would rather make this Instance specific ...

**Review by kulinseth — CHANGES_REQUESTED** [review]

File: aten/src/ATen/native/mps/MPSGraphVenturaOps.h [hunk 2]
```diff
@@ -89,4 +89,26 @@ typedef NS_ENUM(NSUInteger, MPSGraphResizeNearestRoundingMode)
                                     scaleOffsetTensor:(MPSGraphTensor * _Nonnull) scaleOffset
                                                layout:(MPSGraphTensorNamedDataLayout) layout
                                                  name:(NSString * _Nullable) name;
+
+- (MPSGraphTensor * _Nonnull) sampleGridWithSourceTensor:(MPSGraphTensor * _Nonnull) source
+                               coordinateTensor:(MPSGraphTensor * _Nonnull) coordinates
```
Comment (razarmehr) [inline aten/src/ATen/native/mps/MPSGraphVenturaOps.h:94]: Can you fix the indents? I got this comment with a similar issue:
https://github.com/pytorch/pytorch/pull/91669#discussion_r1061045475

**Review by razarmehr — APPROVED** [review]
Looks good. Just left a minor comment.

File: aten/src/ATen/native/mps/MPSGraphVenturaOps.h (hunk 2 above)
Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/MPSGraphVenturaOps.h:94]: Thanks - I'll update this.

**Pull request merged**